src/main.rs

```diff
@@ -1,100 +1,119 @@
+use anyhow::Result;
 use aptos_crypto::ed25519::{Ed25519PrivateKey, Ed25519PublicKey};
 use aptos_keygen::KeyGen;
 use aptos_types::{
     account_address::{create_multisig_account_address, AccountAddress},
     transaction::authenticator::AuthenticationKey,
 };
-use std::env;
-use std::thread;
-use std::time::Duration;
+use clap::Parser;
+use crossbeam::channel;
+use std::thread::{self, available_parallelism};
 
-const PREFIX_ARG: usize = 1;
-const N_THREADS_ARG: usize = 2;
-const MULTISIG_ARG: usize = 3;
-const MULTISIG_FLAG: &str = "-m";
-const SEQUENCE_NUMBER: u64 = 0;
+#[derive(Parser, Debug)]
+struct CliArgs {
+    // Number of vanity address keypairs to generate
+    #[arg(short, long)]
+    number: u8,
```

Comment:
We will probably want this as a `u64` for benchmarking/if service providers want to get a batch set etc.